### PR TITLE
Update methods (and a fix to where)

### PR DESCRIPTION
**Pull request by bmneely**

### CODE REVIEW

**Comment by bmneely** [comment]
@robmadden want to take a look at this as well?

File: lib/bloc_record/persistence.rb [hunk 1]
```diff
@@ -21,6 +21,25 @@ def create(attrs)
...
     end
+
+    def update(id, updates)
+      updates = BlocRecord::Utility.convert_keys(updates)
+      updates.delete "id"
+      updates_array = updates.map { |key, value| "#{key}=#{BlocRecord::Utility.sql_strings(value)}" }
+
+      where_clause = id.nil? ? ";" : "WHERE id = #{id};"
+
+      connection.execute <<-SQL
```
Comment (robmadden) [inline lib/bloc_record/persistence.rb]: Is `connection.execute` atomic?

File: lib/bloc_record/persistence.rb [hunk 2]
```diff
@@ -21,6 +21,25 @@ def create(attrs)
...
+        UPDATE #{table}
+        SET #{updates_array * ","} #{where_clause}
+      SQL
+
+      true
```
Comment (robmadden) [inline lib/bloc_record/persistence.rb]: Do we need exception handling and return false if an exception is thrown?

Comment (bmneely) [inline lib/bloc_record/persistence.rb]: We live on the edge, without exception handling!

(It's left for the assignment)

File: lib/bloc_record/selection.rb [hunk 3]
```diff
@@ -85,6 +85,70 @@ def all
...
+          SELECT #{columns.join ","} FROM #{table}
+          WHERE #{expression};
+        SQL
+      when Hash
+        expression_hash = args.first
+        expression_hash = BlocRecord::Utility.convert_keys(expression_hash)
+        expression = expression_hash.keys.map {|key| "#{key}=:#{key}"}.join(" and ")
+        sql = <<-SQL
+          SELECT #{columns.join ","} FROM #{table}
+          WHERE #{expression};
```
Comment (robmadden) [inline lib/bloc_record/selection.rb]: Lines 91, 100, and 108 are redundant, but I'm not sure how you would make it more DRY.

Comment (robmadden) [inline lib/bloc_record/selection.rb]: Make a small method that takes three params...?

**Comment by robmadden** [comment]
Overall, this is impressive. Only thing, do we want comments speckled throughout this for some rationale?

File: lib/bloc_record/persistence.rb (hunk 1 above)
Comment (bmneely) [inline lib/bloc_record/persistence.rb]: I'm not sure, and the docs don't really say...

File: lib/bloc_record/selection.rb (hunk 3 above)
Comment (bmneely) [inline lib/bloc_record/selection.rb]: Yeah, if they weren't inside a SQL string I'd probably prioritize refactoring them, but given that they're all inside statements, I think that would probably make things more confusion.

**Comment by bmneely** [comment]
Comments probably wouldn't hurt, but given that the whole thing is going to be sliced up by Aaron and explained in the checkpoints, maybe not necessary?

**Pull request merged**